pages/api/auth/[...nextauth].tsx

```diff
@@ -1,10 +1,134 @@
 import NextAuth from "next-auth";
-import Providers from "next-auth/providers";
+import Providers, { AppProviders } from "next-auth/providers";
 import { authenticator } from "otplib";
 
-import { ErrorCode, Session, verifyPassword } from "@lib/auth";
+import { ErrorCode, isGoogleLoginEnabled, Session, verifyPassword } from "@lib/auth";
 import { symmetricDecrypt } from "@lib/crypto";
 import prisma from "@lib/prisma";
+import { isSAMLLoginEnabled, samlLoginUrl } from "@lib/saml";
+import slugify from "@lib/slugify";
+
+import { IdentityProvider } from ".prisma/client";
+
+function randomString(length = 12) {
+  let result = "";
+  const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  const charactersLength = characters.length;
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+}
```

Comment:
We already have a randomString function in `playwright/lib/testUtils.ts` Let's use that or relocate it to a more generic place if necessary.

Reply:
I'll move it to lib